### PR TITLE
Gateway UI Login

### PR DESCRIPTION
**Pull request by okjodom**
Reuses `Login` component experience from guardian UI for Gateway UI login. When a user first starts a browser session, they are challenged to provide  a password to log in. Gateway UI attempts to request `/info` as a way of checking the validity of this password and if successful, the password is saved persisted through the extent of the browser session.

**Note**: Gateway UI still supports password configuration through `REACT_APP_FM_GATEWAY_PASSWORD`. When this is set, the login screen is skipped all together

### CODE REVIEW

**Comment by wbobeirne** [comment]
Sorry Jodom, lots of component conflicts from #240 😢

**Review by wbobeirne — COMMENTED** [review]
It's not breaking, but I'd like to see if we could fix the issue with the login flashing. Users on slower networks may be confused seeing the login screen come up for longer times before it flips over to the auth'd state.

**Comment by wbobeirne** [comment]
@okjodom I fixed the conflicts but had to force push due to rebase, sorry for the annoyance.

**Comment by okjodom** [comment]
> @okjodom I fixed the conflicts but had to force push due to rebase, sorry for the annoyance.

No worries :)

**Review by wbobeirne — APPROVED** [review]
@okjodom I pushed fixes for the notes I made, if the fixes look good to you then this is good to go!

**Comment by okjodom** [comment]
Testing now

File: apps/guardian-ui/src/App.tsx [hunk 1]
```diff
@@ -29,7 +35,13 @@ export const App = React.memo(function App() {
     if (state.needsAuth) {
       return (
         <Wrapper>
-          <Login />
+          <Login
+            checkAuth={(password) => api.testPassword(password || '')}
+            setAuthenticated={() =>
+              dispatch({ type: APP_ACTION_TYPE.SET_NEEDS_AUTH, payload: false })
+            }
+            parseError={formatApiErrorMessage}
```
[the remark this answers is not in the document]
Comment (okjodom) [reply apps/guardian-ui/src/App.tsx:43]: 💯

**Comment by okjodom** [comment]
tested ack ✅ ✅ Thanks for the rebase and fixes, @wbobeirne :)

**Pull request merged**